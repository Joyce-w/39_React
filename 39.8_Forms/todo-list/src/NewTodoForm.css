.NewTodoForm-form{
    margin: 2% auto;
    width: fit-content;
    padding: 1%;
    border: 3px solid grey;
    border-radius: 15px;
    box-shadow: 4px 4px 0 rgba(0, 0, 0, 0.3)
}

.NewTodoForm-btn{
    margin:0.5em;
    border-radius: 15px;
    border: solid grey 1px;
    background-color: rgb(226, 226, 226);
}

.NewTodoForm-input{
    border-radius: 5px;
    border: solid grey 1px;
}