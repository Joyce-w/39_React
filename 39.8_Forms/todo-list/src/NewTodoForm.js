import React, { useState } from "react";
import './NewTodoForm.css'


const TodoListForm = ({updateTasks}) => {

    const [task, setTask] = useState('')
    console.log(task)
    //handle submit for form
    const handleSubmit = (e) => {
        e.preventDefault();
        updateTasks(task)
        setTask('')
    }

    //handle input change and update to refelect in state
    const handleChange = (e) => {
        const { value } = e.target
        setTask(value)

    }

    return (
        <>
            <form className="NewTodoForm-form" onSubmit={ handleSubmit }>
                <label htmlFor="task">New Todo </label>
                <input
                    className="NewTodoForm-input"
                    id="task"
                    name='task'
                    type="text"
                    placeholder="More things to do"
                    value={task}
                    onChange={ handleChange }
                />
                <button className="NewTodoForm-btn">Add new task</button>
            </form>
        </>
    )
}

export default TodoListForm;