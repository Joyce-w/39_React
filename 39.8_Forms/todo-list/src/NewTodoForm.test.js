import react from "react"
import { render } from '@testing-library/react';
import NewTodoForm from "./NewTodoForm"

//smoketest
test('renders learn react link', () => {
  render(<NewTodoForm />);
});

//snapshot
it("matches snapshot", () => {
  const { asFragment } = render(<NewTodoForm />)
  expect(asFragment()).toMatchSnapshot();
})

//specialized test
test("it should display a form", () => {
  const { getByLabelText, getByText} = render(<NewTodoForm />)
  
  expect(getByLabelText("New Todo")).toBeInTheDocument("task");
  expect(getByText("Add new task")).toHaveClass("NewTodoForm-btn");

})