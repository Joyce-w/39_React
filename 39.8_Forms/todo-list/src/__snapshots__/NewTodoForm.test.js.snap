// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm-form"
  >
    <label
      for="task"
    >
      New Todo 
    </label>
    <input
      class="NewTodoForm-input"
      id="task"
      name="task"
      placeholder="More things to do"
      type="text"
      value=""
    />
    <button
      class="NewTodoForm-btn"
    >
      Add new task
    </button>
  </form>
</DocumentFragment>
`;
